nomos-core/chain-defs/src/header/mod.rs

```diff
@@ -147,6 +147,19 @@ impl From<HeaderId> for [u8; 32] {
     }
 }
 
+impl TryFrom<&[u8]> for HeaderId {
+    type Error = ();
```

Comment:
`()` doesn't implement error which will make it difficult later if we have to use `?`. Even something generic would be better. We could return the length of the input as a hint on where we fucked up :sweat_smile:

Reply:
yes, either use infallible if it can't fail or some real error in this case when there's a length mismatch